src/main/java/de/tum/in/www1/artemis/domain/SubmissionVersion.java

```diff
@@ -0,0 +1,118 @@
+package de.tum.in.www1.artemis.domain;
+
+import java.time.Instant;
+
+import javax.persistence.*;
+
+import org.hibernate.annotations.Cache;
+import org.hibernate.annotations.CacheConcurrencyStrategy;
+import org.springframework.data.annotation.CreatedDate;
+import org.springframework.data.annotation.LastModifiedDate;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+
+@Entity
+@Table(name = "submission_version")
+@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
+@EntityListeners(AuditingEntityListener.class)
+public class SubmissionVersion {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne
+    private Submission submission;
+
+    @ManyToOne
+    private User author;
+
+    @Lob
+    @Column(name = "content")
+    private String content;
+
+    @CreatedDate
+    @Column(name = "created_date", updatable = false)
+    private Instant createdDate = Instant.now();
+
+    @LastModifiedDate
+    @Column(name = "last_modified_date")
+    private Instant lastModifiedDate = Instant.now();
```

Comment:
add a small comment that this is set automatically by spring / hibernate

Reply:
Done